app/src/main/java/org/sil/storyproducer/model/StoryIO.kt

```diff
@@ -39,41 +43,54 @@ fun storyFromJson(context: Context, storyTitle: String): Story?{
                 ?: return null
         return adapter.fromJson(fileContents)
     } catch (e: Exception) {
-        return null
+        throw StoryIOException("StoryIO: storyFromJson() parse exception from moshi")
     }
 }
 
-fun parseStoryIfPresent(context: Context, storyPath: androidx.documentfile.provider.DocumentFile): Story? {
+/**
+ * Attempt to parse a story from directory. A null Story is returned in the case of a parse error or non-template directory
+ *
+ * @return Story
+ */
+fun parseStoryIfPresent(context: Context, storyPath: DocumentFile): Story? {
+    if(!storyPath.isDirectory || storyPath.name == WORD_LINKS_DIR) return null
+
     var story: Story?
-    //Check if path is path
-    if(!storyPath.isDirectory) return null
-    //make a project directory if there is none.
     if (storyRelPathExists(context,PROJECT_DIR,storyPath.name!!)) {
-        //parse the project file, if there is one.
-        story = storyFromJson(context, storyPath.name!!)
-        //if there is a story from the file, do not try to read any templates, just return.
-        if(story != null) return story
-    }
-    try {
-        story = parsePhotoStoryXML(context, storyPath)
-    } catch (e : Exception){
-        FirebaseCrashlytics.getInstance().recordException(e)
-        story = null
-    }
-    if(story == null){
+        // project directory exists, so working on a JSON template
         try {
-            story = parseBloomHTML(context, storyPath)
+            story = storyFromJson(context, storyPath.name!!)
+            return story
+        } catch (e: Exception) {
+            FirebaseCrashlytics.getInstance().recordException(e)
+            return null
+        }
```

Comment:
If we only return a null or Story from "storyFromJson" (eliminating the throw in storyFromJson).  We can replace these six new lines of code with: return(storyFromJson(context, storyPath.name!!)).  

I think to simplify the code, routines like parsePhotoStoryXMLn & parseBloomHTML should only return "Story or null".    This way we don't clutter up this code with try/catch blocks.  This would be an improvement later on.